src/store/slices/productsFilterSlice/slice.ts

```diff
@@ -0,0 +1,37 @@
+import { ProductsFilterState } from "@/common/types";
+import { PayloadAction, createSlice } from "@reduxjs/toolkit";
+
+const initialState: ProductsFilterState = {
+  searchValue: "",
+  productCategory: "",
+  productBrand: "",
+  isPending: false,
+  error: null,
+};
+
+export const productsFilterSlice = createSlice({
+  name: "products_filter",
+  initialState,
+  reducers: {
+    setSearchValue: (state, action: PayloadAction<string>) => {
+      return {
+        ...state,
+        searchValue: action.payload,
+      };
+    },
+    setCategory: (state, action: PayloadAction<string>) => {
+      return {
+        ...state,
+        productCategory: action.payload,
+      };
+    },
+    setBrand: (state, action: PayloadAction<string>) => {
+      return {
+        ...state,
+        productBrand: action.payload,
+      };
+    },
+  },
+});
```

Comment:
In RTK there is no need to return new state object. 
It will recognize changes even if a single field is being updated. 

=====================

setSearchValue: (state, action: PayloadAction<string>) => {
   state.searchValue = action.payload;
    },
 setCategory: (state, action: PayloadAction<string>) => {
   state. productCategory = action.payload;   
    },

 setBrand: (state, action: PayloadAction<string>) => {
      state.productBrand =  action.payload,
    },

===============

The example above will do the same.